### DISCUSSION TITLE
Container issue Using Docker with Windows 10

### DISCUSSION THREAD

**Discussion by 404GoatFound** [teslamate-org/teslamate#2598] [Q&A]
Seems that my teslamate running on port 4000 and my grafana running on port 3000 are running into issues right after my install. I've attached snippets of the logs. Is this a port issue? No other applications are using the ports. I can get to grafana UI through localhost but not teslamate. Any ideas?
![Capture 1](https://user-images.githubusercontent.com/103010775/161664723-048c6eab-138c-46af-bf0a-872e3d649691.PNG)
![Capture 2](https://user-images.githubusercontent.com/103010775/161664724-df8cd7da-1153-4fbe-9ce2-a90f2cefd968.PNG)

**Comment by 404GoatFound** [comment]
![Capture 3](https://user-images.githubusercontent.com/103010775/161664827-fff63abc-c4b3-460b-9a75-5ccaa3c25c41.PNG)
Another snip showing all containers.

**Comment by CrowMash** [comment]
Looks like there is an issue with your password, can you check your docker-compose.yml file? Some users had issue when their password started with a space, maybe that's that.

**Reply by 404GoatFound** [reply]
I thought the same, but that wasn't the issue. I don't even get the chance to input a password? The docker container just continuously reboots. I'll attach my .YML file when I get chance shortly.

**Reply by 404GoatFound** [reply]
version: "3"

services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: always
    environment:
      - ENCRYPTION_KEY=1234
      - DATABASE_USER=teslamate
      - DATABASE_PASS=password
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
      - MQTT_HOST=mosquitto
    ports:
      - 4000:4000
    volumes:
      - ./import:/opt/app/import
    cap_drop:
      - all

  database:
    image: postgres:13
    restart: always
    environment:
      - POSTGRES_USER=teslamate
      - POSTGRES_PASSWORD=password
      - POSTGRES_DB=teslamate
    volumes:
      - teslamate-db:/var/lib/postgresql/data

  grafana:
    image: teslamate/grafana:latest
    restart: always
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=password
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
    ports:
      - 3001:3001
    volumes:
      - teslamate-grafana-data:/var/lib/grafana

  mosquitto:
    image: eclipse-mosquitto:2
    restart: always
    command: mosquitto -c /mosquitto-no-auth.conf
    # ports:
    #   - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

volumes:
  teslamate-db:
  teslamate-grafana-data:
  mosquitto-conf:
  mosquitto-data:

**Reply by 404GoatFound** [reply]
Is an encryption key required? I've tried leaving it blank as well. Maybe that's the issue?
